### PR TITLE
Update `make docs` procedure and related workflows

### PR DESCRIPTION
**Pull request by jdbaldry**
Add a workflow to automatically update the procedure daily. 
I've added myself as CODEOWNER via the @grafana/docs-tooling role to ensure that I can be notified and promptly review the PRs to update the workflow in the future.

I'll need an admin to give the team @grafana/docs-tooling write access to the repository.

### CODE REVIEW

**Comment by tolzhabayev** [comment]
I can help you with write access to the repo

File: .github/workflows/update-make-docs.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          BRANCH="$(date +%Y-%m-%d)/update-make-docs"
+          git checkout -b "${BRANCH}"
+          git add .
+          git config --local user.email "bot@grafana.com"
+          git config --local user.name "grafanabot"
+          git commit -m "Update \`make docs\` procedure"
+          git push -v origin "refs/heads/${BRANCH}"
+          gh pr create --fill
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tolzhabayev) [inline .github/workflows/update-make-docs.yml:27]: I am not quite sure you will have the write permissions by default in this token?

You might need to add:
```
    permissions:
      contents: write
```

**Review by tolzhabayev — APPROVED** [review]

File: .github/workflows/update-make-docs.yml (hunk 1 above)
Comment (jdbaldry) [reply .github/workflows/update-make-docs.yml:27]: Is there something specific to this repository I need to be aware of? This workflow works in https://github.com/grafana/technical-documentation/actions/workflows/update-make-docs.yml.

**Comment by tolzhabayev** [comment]
Please also take a look at the results of codeowners validator: 
https://github.com/grafana/grafana/actions/runs/5541866703/jobs/10115829692?pr=71537

File: .github/workflows/update-make-docs.yml (hunk 1 above)
Comment (tolzhabayev) [reply .github/workflows/update-make-docs.yml:27]: Should not be the case, so let's see :)

Comment (jdbaldry) [reply .github/workflows/update-make-docs.yml:27]: 🤞

**Pull request merged**